scss/_patterns_modal.scss

```diff
@@ -53,20 +56,22 @@
   }
 
   .p-modal__close {
+    $icon-size: map-get($icon-sizes, default);
+
     background: {
       image: url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' height='90' width='90'%3E%3Cg color='%23000'%3E%3Cpath fill='none' d='M0 0h90v90H0z'/%3E%3Cpath d='M14.52 6L6 14.52 36.48 45 6 75.49 14.52 84 45 53.52 75.48 84 84 75.49 53.52 45 84 14.52 75.48 6 45 36.49z' fill='%23888'/%3E%3C/g%3E%3C/svg%3E");
       position: center;
       repeat: no-repeat;
-      size: $sp-medium;
+      size: $icon-size;
     }
 
     border: 0;
     box-sizing: content-box;
-    height: $sp-medium;
-    margin: -#{$sp-medium} -#{$sp-medium} 0 0;
-    padding: $sp-medium;
+    height: $icon-size;
+    margin: -#{$icon-size} -#{$icon-size} 0 0;
```

Comment:
Since there's a border under the title, the button is identified as part of a title bar. As such, it would be better to position it so it aligns with the title. I'll add a screenshot as a regular comment, the code for I used (hardcoded rem values just for testing, but it should use the var for modal padding):
```{
    position: absolute;
    right: 1rem;
    top: .5rem;
}
```